website/docs/r/org_vdc_template_instance.html.markdown

```diff
@@ -0,0 +1,123 @@
+}
+
+data "vcd_network_pool" "np1" {
+  name = "NSX-T Overlay 1"
+}
+
+resource "vcd_org_vdc_template" "tmpl" {
+  name               = "myTemplate"
+  tenant_name        = "myAwesomeTemplate"
+  description        = "Requires System privileges"
+  tenant_description = "Any tenant can use this"
+  allocation_model   = "AllocationVApp"
+
+  compute_configuration {
+    cpu_limit         = 0
+    cpu_guaranteed    = 20
+    cpu_speed         = 256
+    memory_limit      = 1024
+    memory_guaranteed = 30
+  }
+
+  provider_vdc {
+    id                  = data.vcd_provider_vdc.pvdc1.id
+    external_network_id = data.vcd_external_network_v2.ext_net.id
+  }
+
+  provider_vdc {
+    id                  = data.vcd_provider_vdc.pvdc2.id
+    external_network_id = data.vcd_external_network_v2.ext_net.id
+  }
+
+  storage_profile {
+    name    = "*"
+    default = true
+    limit   = 1024
+  }
+
+  network_pool_id = data.vcd_network_pool.np1.id
+
+  readable_by_org_ids = [
+    data.vcd_org.org.id
+  ]
+}
+
+resource "vcd_org_vdc_template_instance" "my_instance" {
+  org_vdc_template_id = vcd_org_vdc_template.tmpl.id
+  name                = "myInstantiatedVdc"
+  description         = "A new VDC"
+  org_id              = data.vcd_org.org.id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org_vdc_template_id` - (Required) The ID of the VDC Template to instantiate
+* `name` - (Required) Name to give to the instantiated Organization VDC
+* `description` - (Optional) Description of the instantiated Organization VDC
+* `org_id` - (Required) ID of the Organization where the VDC will be instantiated
+
+## Attribute Reference
+
+There are no read-only attributes. However, after the `vcd_org_vdc_template_instance` resource is created successfully,
+the identifier of the new VDC is saved in the Terraform state as the `vcd_org_vdc_template_instance` resource ID.
+
+## How to manage the VDC instantiated from a template using Terraform
+
+If users want to modify the new instantiated VDC, they can [import](/providers/vmware/vcd/latest/docs/guides/importing_resources#semi-automated-import-terraform-v15) it.
+In the same `.tf` file (once the VDC has been instantiated), or in a new one, we can place the following snippet: 
+
+```hcl
+import {
+  to = vcd_org_vdc.imported
+  id = "my_org.myInstantiatedVdc" # Using the same names from the example
+}
+```
+
+Note that this importing mechanism still does not support `${}` placeholders, so the Organization and VDC name must be explicitly
+written. When running the `terraform plan -generate-config-out=generated_resources.tf`, Terraform will generate the new file
+`generated_resources.tf` with the instantiated VDC code.
+
+With a subsequent `terraform apply`, the instantiated VDC will be managed by Terraform as a normal `vcd_org_vdc` resource.
+
+Please take into account that deleting the `vcd_org_vdc_template_instance` resource will attempt to **delete the instantiated VDC** it created,
```

Comment:
This baffles me a bit. We just described a whole section on how to manage template-instantiated VCD by importing. But then tell that if you delete `vcd_org_vdc_template_instance`, it will delete the instantiated VDC.

This feels a bit asymmetrical (and dangerous). Should it really delete the VDC instance if modifying it doesn't work?

CC: @Didainius @dataclouder

Reply:
Added three new arguments:
- `delete_instantiated_vdc_on_removal`
- `delete_force`
- `delete_recursive`
Updated docs and code. I'll run a few tests.